src/tests/system/tests/test_netgroups.py

```diff
@@ -309,3 +312,112 @@ def test_netgroup__uid_gt_2147483647(client: Client, provider: GenericProvider):
         result = client.tools.getent.group(grpname)
         assert result is not None, f"getent group for group '{grpname}' is empty!"
         assert result.name == grpname, f"Group name '{grpname}' did not match result '{result.name}'!"
+
+
+@pytest.mark.importance("low")
+@pytest.mark.ticket(bz=1576852)
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_netgroup__nss_responder(client: Client, provider: GenericProvider):
+    """
+    :title: SSSD nss responder handles correctly netgroup timeout when backend is offline
+    :setup:
+        1. A user (user-1) and a netgroup (ng-1) are created, and the user is added as a member of the netgroup
+        2. SSSD is restarted with a clean configuration to ensure no stale data is present
+    :steps:
+        1. Depending on the type of provider (AD, IPA, Samba, or LDAP), the script updates the SSSD configuration
+            with an incorrect server URI (e.g., typo.dc.hostname)
+        2. SSSD is restarted again to apply the new configuration
+        3. Checks the status of the SSSD domain
+        4. Capture the process ID (PID) of the sssd_nss process
+        5. Try to retrieve the netgroup information again, expecting it to fail since the SSSD domain is offline
+        6. Verify that the SSSD processes (sssd, sssd_be, sssd_nss, sssd_pam) are still running and that the
+            sssd_nss process ID has not changed, indicating that SSSD has not crashed or restarted unexpectedly
+    :expectedresults:
+        1. SSSD configured with incorrect server uri
+        2. SSSD restarted
+        3. SSSD domain is offline
+        4. Pid of sssd_nss captured
+        5. Netgroup info can't be retrieved
+        6. SSSD nss responder has the same pid as before
+    :customerscenario: True
+    """
+    user = provider.user("user-1").add()
+    netgroup = provider.netgroup("ng-1").add().add_member(user=user)
+    client.sssd.restart(clean=True)
+
+    hostname = client.host.hostname
+    if isinstance(provider, (AD)) or isinstance(provider, (Samba)):
+        bad_ldap_uri = "typo.dc.%s" % hostname
+        client.sssd.dom("test").update(ad_server=bad_ldap_uri)
+
+    elif isinstance(provider, (IPA)):
+        bad_ldap_uri = "typo.master.%s" % hostname
+        client.sssd.dom("test").update(ipa_server=bad_ldap_uri)
+
+    elif isinstance(provider, (LDAP)):
+        bad_ldap_uri = "ldaps://typo.%s" % hostname
+        client.sssd.dom("test").update(ldap_uri=bad_ldap_uri)
+
+    client.sssd.restart(clean=True)
+
+    # Check backend status
+    assert "Offline" in client.host.conn.run("sssctl domain-status test -o").stdout
```

Comment:
`sssctl` utils already provides a way to run this command from the framework itself. Check https://github.com/SSSD/sssd-test-framework/blob/master/sssd_test_framework/utils/sssctl.py#L232 for additional details on how to do it.